src/Console/src/Configurator/Attribute/Parser.php

```diff
@@ -179,12 +179,15 @@ private function typecast(mixed $value, \ReflectionProperty $property): mixed
             return $value;
         }
 
-        return match ($type->getName()) {
-            'int' => (int) $value,
-            'string' => (string) $value,
-            'bool' => (bool) $value,
-            'float' => (float) $value,
-            'array' => (array) $value,
+        $typeName = $type->getName();
+
+        return match (true) {
+            $typeName === 'int' => (int) $value,
+            $typeName === 'string' => (string) $value,
+            $typeName === 'bool' => (bool) $value,
+            $typeName === 'float' => (float) $value,
+            $typeName === 'array' => (array) $value,
+            \enum_exists($typeName) => $typeName::from($value),
```

Comment:
If the console option receives an incorrect value, not allowed by the PHP enum, we should display an error to the user like this: 'Wrong option value. Allowed options: 'foo', 'bar'.